src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -24,7 +25,7 @@ namespace detail {
 // passed as a void* and re-interpreted appropriately.
 template <typename IntegrandType>
 double integrand(const double x, void* const params) {
-  const auto function = reinterpret_cast<IntegrandType*>(params);
+  const auto function = reinterpret_cast<IntegrandType* const>(params);
```

Comment:
I think the fact that this change does not produce a warning is a bug in GCC.  (It warns in normal functions but not in function templates, even if none of the relevant types are dependent.)  In a normal function this would give "warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]".